### PR TITLE
feat: cli reference & configuration

### PR DESCRIPTION
**Pull request by chloeminkyung**
feat: cli reference

### CODE REVIEW

File: docs/cli-reference.md [hunk 1]
```diff
@@ -1 +1,123 @@
...
+Dozer Command Line Interface is a unified tool and exposes commands for publishing and managing your applications, as well as lower level tools to interact with each component of your application. This reference guides you through getting started with the CLI.
+
+## Synopsis
+```
+dozer [options] <command> <subcommand> [parameters]
+```
+Use `dozer help`, `dozer -h` or `dozer --help` to view a list of available commands. Use `dozer <command> help`, `dozer <command> -h` or `dozer <command> --help` for information on a specific command with available subcommands. The synopsis for each command shows its parameters and their usage. Optional parameters are shown in square brackets.
+
+
+## Basic Commands
```
Comment (snork-alt) [inline docs/cli-reference.md:11]: I would distinguish between commands used in teh development lifecycle (like init, live, clean, etc) and commands used for deployments (like run). Mention that dozer has an app (pipeline) component and an api component and those can be run together or separately. Running together is generally used during development while running them separately is used during deployment. Then we point to a deployment page

**Review by snork-alt — CHANGES_REQUESTED** [review]

File: docs/cli-reference.md [hunk 2]
```diff
@@ -1 +1,123 @@
...
+> 
+> Provide user profile to cloud login for deployment.
+
+## Global Options
+
+> `--config-path (string)`, `-c (string)`
+> 
+> Provide dozer configuration YAML file with given PATH.
+> 
+> `--config-overrides (string)`
```
[the remark this answers is not in the document]
Comment (chloeminkyung) [reply docs/cli-reference.md:115]: addressed

File: docs/cli-reference.md [hunk 3]
```diff
@@ -1 +1,123 @@
...
+```
+
+and make sure you add following part in the dozer configuration file.
+```yaml
+api:
+  api_security:
+    !Jwt
+```
+
+### Dozer Cloud
```
[the remark this answers is not in the document]
Comment (chloeminkyung) [reply docs/cli-reference.md:89]: How do you want to order to be? if it's under Cloud deployment command header then what should we do

File: docs/cli-reference.md (hunk 1 above)
Comment (chloeminkyung) [reply docs/cli-reference.md]: addressed

**Review by snork-alt — CHANGES_REQUESTED** [review]
In the configuration we should list:
- all the global configuration parameters and the meaning of each one ( for example, max map size, monitoring, etc). Whatever is global
- for endpoints we should also explain the config parameters for the endpoint type. For example secondary indexes, etc etc
- for SQL we should explain that every SQL statement required an INTO that represents the cache
-

File: docs/cli-reference.md [hunk 4]
```diff
@@ -1 +1,228 @@
...
+> |  _ \ / _ \__  / ____|  _ \
+> | | | | | | |/ /|  _| | |_) |
+> | |_| | |_| / /_| |___|  _ <
+> |____/ \___/____|_____|_| \_\
+> 
+> INFO Starting live server
+> ```
+
+### `dozer`
+If you run Dozer CLI with no command are passed like below, dozer will bring up both `app` and `api` services.
```
Comment (snork-alt) [inline docs/cli-reference.md:40]: Say also, app is the processing pipeline transforming the data, while API is the Api server that serves gRPC and REST requests

File: docs/cli-reference.md [hunk 5]
```diff
@@ -1 +1,228 @@
...
+>  ...
+>  INFO Created new build v0001
+> ```
+
+### `dozer run app`
+If you run Dozer CLI with `run` command with `app` subcommand like below, dozer will bring up app service.
+
+### `dozer run api`
+If you run Dozer CLI with `run` command with `api` subcommand like below, dozer will bring up api service.
+
```
Comment (snork-alt) [inline docs/cli-reference.md:63]: Add ..for more details on self-hosting click here. And we link to the deployment section

File: docs/cli-reference.md [hunk 6]
```diff
@@ -1 +1,228 @@
...
+> ```
+
+#### `dozer security generate-token`
+If you want to utilize api security features, run following command to generate the token. Make sure you add following part in the dozer configuration file.
+```yaml
+api:
+  api_security:
+    !Jwt
+```
+
```
Comment (snork-alt) [inline docs/cli-reference.md:169]: Add: for mode details on how.to.configure authentication click here ...and we link.to the Api security section

File: docs/configuration.md [hunk 7]
```diff
@@ -1 +1,78 @@
...
+| **`endpoints`**   | Array  | true      | `endpoints: `<br/>&nbsp;&nbsp;&nbsp;&nbsp;`- name: trips` <br/>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`path: /trips` <br/>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`table_name: trips`<br/>more example available [here](#api-endpoints)                                                                                                                                                                                                                                                                                                                                                                                                                                       |
+
+### Conventions
```
Comment (snork-alt) [inline docs/configuration.md:27]: No need to specify conventions..this is standard yaml syntax

File: docs/configuration.md [hunk 8]
```diff
@@ -1 +1,78 @@
...
+- Port number
+- Database name
+- Username and password
+- Authentication method (if applicable)
+- Data warehouse or storage service credentials (if applicable)
+- API keys or tokens (if applicable)
+
+## SQL Transformations
+
+Define the data transformation rules you want to apply to the ingested data.
```
Comment (snork-alt) [inline docs/configuration.md:58]: Mention about the INTO like we said

File: docs/configuration.md [hunk 9]
```diff
@@ -1 +1,78 @@
...
+## API Endpoints
+
+Configure the Store/API nodes to expose gRPC and REST APIs for data access. Define the endpoints and routes that users or downstream systems will use to query and retrieve data.
+
+## Global settings (Optional)
+
+| Property                               | Details                                                      | Type    | Default Value | Example                              |
+|----------------------------------------|--------------------------------------------------------------|---------|---------------|--------------------------------------|
+| `app_buffer_size`                      | Pipeline buffer size                                         | Integer | 20000         | `app_buffer_size: 50000`             |
+| `commit_size`                          |                                                              | Integer | 10000         | `commit_size: 100`                   |
```
Comment (snork-alt) [inline docs/configuration.md:70]: We should explain what is it

File: docs/configuration.md [hunk 10]
```diff
@@ -1 +1,78 @@
...
+- Authentication method (if applicable)
+- Data warehouse or storage service credentials (if applicable)
+- API keys or tokens (if applicable)
+
+## SQL Transformations
+
+Define the data transformation rules you want to apply to the ingested data. 
+These SQLs might involve joining tables, aggregating data, or applying filters.
+
+## API Endpoints
```
Comment (snork-alt) [inline docs/configuration.md:61]: We need to specify the meaning of each parameter of endpoint configuration. Like we said primary, secondary, full text indexes, the group by thing

File: docs/configuration.md [hunk 11]
```diff
@@ -1 +1,78 @@
...
+The spec of the configuration file must use these conventions.
+
+- Use **`[]`** to indicate an empty list
+- Use **`null`** to indicate a null value
+- Use **`true`** and **`false`** as only options on boolean fields
+
+
+## Data Sources
+
+Followings are dozer supported connectors and detailed example configurations are available:
```
Comment (snork-alt) [inline docs/configuration.md:37]: We need to specify here we have connections and sources and what is the meaning

**Review by snork-alt — CHANGES_REQUESTED** [review]

File: docs/configuration.md (hunk 9 above)
Comment (chloeminkyung) [reply docs/configuration.md:106]: I will fill up the missing description by getting from the person who implemented it - give me sometime!

File: docs/configuration.md [hunk 12]
```diff
@@ -1 +1,78 @@
...
+| `log_max_num`<br/>`_immutable_entries` | Max number of immutable entries                              | Integer | 100           | `log_max_num_immutable_entries: 100` |
+| `log_storage`                          | The storage to use for the log                               |         |               | `log_storage: `                      |
+| `error_threshold`                      | How many errors we can tolerate before bringing down the app | Integer | 0             | `error_threshold: 10`                |
+| `home_dir`                             | Home directory of dozer application                          | String  | `./.dozer`    | `home_dir: ./.dozer`                 |
+| `cache_dir`                            | Cache directory of dozer application                         | String  | `./.dozer`    | `cache_dir: ./.dozer`                |
+| `cache_max_map_size`                   |                                                              | Integer | `./.dozer`    | `cache_max_map_size: 8589934592`     |
```
[the remark this answers is not in the document]
Comment (chloeminkyung) [reply docs/configuration.md:114]: I will fill up the missing description by getting from the person who implemented it

**Review by snork-alt — APPROVED** [review]

**Pull request merged**